### PR TITLE
Sagoswami/update bike share v2

### PR DESCRIPTION
**Pull request by SamGos93**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 1]
```diff
@@ -0,0 +1,1041 @@
...
+    "plt.title(f\"Predicted vs. Actuals\")\n",
+    "plt.legend([\"actual\", \"forecast\"])\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For more details on what metrics are included and how they are calculated, please refer to [supported metrics](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-understand-automated-ml#regressionforecasting-metrics). You could also calculate residuals, like described [here](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-understand-automated-ml#residuals).\n",
```
Comment (vlbejan) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:913]: Please replace "You could also calculate residuals, like described ... " with "You could also calculate residuals as described ..."

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 2]
```diff
@@ -0,0 +1,1041 @@
...
+    "    description=\"this is a sample batch endpoint\",\n",
+    ")\n",
+    "ml_client.begin_create_or_update(endpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create a batch deployment, we will use the rolling_script.py which will load the model and will call the rolling forecast method each time we will envoke the endpoint."
```
Comment (vlbejan) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:725]: The rolling forecast is misleading. We are using rolling evaluation. Please incorporate the writeup from the v1 version of the [notebook](https://github.com/Azure/azureml-examples/blob/main/v1/python-sdk/tutorials/automl-with-azureml/forecasting-bike-share/auto-ml-forecasting-bike-share.ipynb) (Section: Retrieving forecasts from the model).

**Review by vlbejan — CHANGES_REQUESTED** [review]

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "\n",
     "**Motivations** - This notebook explains how to setup and run an AutoML forecasting job. This is one of the nine ML-tasks supported by AutoML. Other ML-tasks are 'regression', 'classification', 'image classification', 'image object detection', 'nlp text classification', etc.\n",
     "\n",
-    "In this example we use the associated New York City energy demand dataset to showcase how you can use AutoML for a simple forecasting problem and explore the results. The goal is predict the energy demand for the next 48 hours based on historic time-series data.\n"
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: I have pushed energy demand notebook by mistake. I made some error in rebasing my branch with main. Thanks!

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 4]
```diff
@@ -533,14 +921,961 @@
...
+    "    description=\"environment for automl inference\",\n",
+    "    image=\"mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04:20210727.v1\",\n",
+    "    conda_file=\"artifact_downloads/outputs/conda_env_v_1_0_0.yml\",\n",
+    ")\n",
+    "\n",
+    "endpoint = BatchEndpoint(\n",
+    "    name=batch_endpoint_name,\n",
+    "    description=\"this is a sample batch endpoint\",\n",
+    ")\n",
+    "ml_client.begin_create_or_update(endpoint)"
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: done in bike share notebook

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 5]
```diff
@@ -533,14 +921,961 @@
...
+    "    endpoint_name=online_endpoint_name,\n",
+    "    model=registered_model.id,\n",
+    "    environment=env,\n",
+    "    code_configuration=code_configuration,\n",
+    "    instance_type=\"Standard_DS2_V2\",\n",
+    "    instance_count=1,\n",
+    ")\n",
+    "# deployment to take 100% traffic\n",
+    "endpoint.traffic = {\"energy-managed-deploy\": 100}\n",
+    "ml_client.begin_create_or_update(endpoint)"
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: done!

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 6]
```diff
@@ -141,81 +207,164 @@
...
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "# Advanced Forecasting Training <a id=\"advanced_training\"></a>\n",
+    "|**frequency**|Forecast frequency. This optional parameter represents the period with which the forecast is desired, for example, daily, weekly, yearly, etc. Use this parameter for the correction of time series containing irregular data points or for padding of short time series. The frequency needs to be a pandas offset alias. Please refer to [pandas documentation](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects) for more information.\n",
+    "|**cv_step_size**|Number of periods between two consecutive cross-validation folds. The default value is \"auto\", in which case AutoMl determines the cross-validation step size automatically, if a validation set is not provided. Or users could specify an integer value.|\n",
+    "|**target_lags**|The target_lags specifies how far back we will construct the lags of the target variable.|\n",
+    "|**target_rolling_window_size**|The target_rolling_window_size specifies the size of a rolling window for aggregated lookback features.|\n",
+    "\n",
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: Added, thanks!

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 7]
```diff
@@ -426,9 +629,43 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 42,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: Not applicable with my new changes

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb [hunk 8]
```diff
@@ -533,14 +921,961 @@
...
+      ]
+     },
+     "execution_count": 60,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ml_client.begin_create_or_update(batch_deployment)"
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-energy-demand/automl-forecasting-task-energy-demand-advanced-mlflow.ipynb]: done

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 9]
```diff
@@ -0,0 +1,1041 @@
...
+    "\n",
+    "**Learning Objectives** - By the end of this tutorial, you should be able to:\n",
+    "- Connect to your AML workspace from the Python SDK\n",
+    "- Create an `AutoML time-series forecasting Job` with the 'forecasting()' factory-fuction\n",
+    "- Train the model using AmlCompute by submitting/running the AutoML forecasting training job\n",
+    "- Obtain the model and use it to generate forecast\n",
+    "\n",
+    "**Motivations** - This notebook explains how to setup and run an AutoML forecasting job. This is one of the nine ML-tasks supported by AutoML. Other ML-tasks are 'regression', 'classification', 'image classification', 'image object detection', 'nlp text classification', etc.\n",
+    "\n",
+    "In this example we use the associated New York City energy demand dataset to showcase how you can use AutoML for a simple forecasting problem and explore the results. The goal is to predict the energy demand for the next 48 hours based on historic time-series data."
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: I made the changes in my local but didn't push it before your review. I have pushed it now. My apologies.

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 10]
```diff
@@ -0,0 +1,1041 @@
...
+    "outputDf.T"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Data\n",
+    "\n",
+    "We will use energy consumption Bike Share Data for model training. The data is stored in a tabular format and includes Bike share data. (To be added)\n",
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: taken care after pushing my local changes

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 11]
```diff
@@ -0,0 +1,1041 @@
...
+    "|**frequency**|Forecast frequency. This optional parameter represents the period with which the forecast is desired, for example, daily, weekly, yearly, etc. Use this parameter for the correction of time series containing irregular data points or for padding of short time series. The frequency needs to be a pandas offset alias. Please refer to [pandas documentation](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects) for more information.\n",
+    "|**cv_step_size**|Number of periods between two consecutive cross-validation folds. The default value is \"auto\", in which case AutoML determines the cross-validation step size automatically, if a validation set is not provided. Or users could specify an integer value.|\n",
+    "|**target_lags**|The target_lags specifies how far back we will construct the lags of the target variable.|\n",
+    "|**target_rolling_window_size**|The target_rolling_window_size specifies the size of a rolling window for aggregated lookback features.|\n",
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: done, thanks

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 12]
```diff
@@ -0,0 +1,1041 @@
...
+    "\n",
+    "### Using lags and rolling window features\n",
+    "This training is also using the **target lags** which are lagged values of the target variable. Since we generate lags with respect to horizon, we therefore must still specify the `forecast_horizon` that the model will learn to forecast. The `target_lags` keyword specifies how far back we will construct the lags of the target variable, and the `target_rolling_window_size` specifies the size of the rolling window over which we will generate the `max`, `min` and `sum` features.\n",
```
[the remark this answers is not in the document]
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: Added

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb (hunk 1 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: my bad, thanks for correcting

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb (hunk 2 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: updated

**Review by vlbejan — APPROVED** [review]

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable [hunk 13]
```diff
@@ -0,0 +1,29 @@
+paths:
+<<<<<<<< HEAD:sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable
```
Comment (vlbejan) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable:2]: Several merge conflicts

**Review by vlbejan — CHANGES_REQUESTED** [review]

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable (hunk 13 above)
Comment (nick863) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable]: Please resolve the conflicts.

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/forecast/forecasting_script.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
+"""
```
Comment (nick863) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/forecast/forecasting_script.py:1]: We do not use this script in the notebook, let us remove it.

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable (hunk 13 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable]: resolved, thanks

**Review by vlbejan — APPROVED** [review]

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 15]
```diff
@@ -0,0 +1,1041 @@
...
+    "from azure.ai.ml import MLClient\n",
+    "from azure.ai.ml.constants import AssetTypes\n",
+    "from azure.ai.ml import automl\n",
+    "from azure.ai.ml import Input\n",
+    "\n",
+    "import json\n",
+    "import logging\n",
+    "from datetime import datetime\n",
+    "import numpy as np\n",
+    "import pandas as pd"
```
Comment (nick863) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:71]: let us remove unused imports:
from datetime import datetime
import numpy as np

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 16]
```diff
@@ -0,0 +1,1041 @@
...
+    "        \"TARGET_COLUMN_NAME\": target_column_name,\n",
+    "    },\n",
+    "    compute = cluster_name,\n",
+    "    instance_count = 2,\n",
+    "    max_concurrency_per_instance = 2,\n",
+    "    mini_batch_size = 10,\n",
+    "    output_action = BatchDeploymentOutputAction.APPEND_ROW,\n",
+    "    output_file_name = output_file,\n",
+    "    retry_settings = BatchRetrySettings(max_retries=3, timeout=30),\n",
+    "    logging_level = \"info\",\n",
```
Comment (nick863) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:753]: logging.INFO

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable (hunk 13 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/data/test-mltable-folder/MLTable]: It was updated in previous commit

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/forecast/forecasting_script.py (hunk 14 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/forecast/forecasting_script.py]: done

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb (hunk 15 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:69]: removed

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb (hunk 16 above)
Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: changed logging_level = logging.INFO

File: .github/workflows/sdk-jobs-automl-standalone-jobs-automl-forecasting-task-bike-share-auto-ml-forecasting-bike-share.yml [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    - name: check out repo
+      uses: actions/checkout@v2
+    - name: setup python
+      uses: actions/setup-python@v2
+      with: 
+        python-version: "3.8"
+    - name: pip install notebook reqs
+      run: pip install -r sdk/python/dev-requirements.txt
+    - name: pip install mlflow reqs
+      run: pip install -r sdk/python/mlflow-requirements.txt
```
Comment (nick863) [inline .github/workflows/sdk-jobs-automl-standalone-jobs-automl-forecasting-task-bike-share-auto-ml-forecasting-bike-share.yml:29]: You will also need forecasting requirements
```
    - name: pip install forecasting reqs
      run: pip install -r sdk/python/forecasting-requirements.txt
```

Comment (SamGos93) [reply .github/workflows/sdk-jobs-automl-standalone-jobs-automl-forecasting-task-bike-share-auto-ml-forecasting-bike-share.yml:29]: added, thanks

File: sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb [hunk 18]
```diff
@@ -0,0 +1,1037 @@
...
+    "        \"TARGET_COLUMN_NAME\": target_column_name,\n",
+    "    },\n",
+    "    compute = cluster_name,\n",
+    "    instance_count = 2,\n",
+    "    max_concurrency_per_instance = 2,\n",
+    "    mini_batch_size = 10,\n",
+    "    output_action = BatchDeploymentOutputAction.APPEND_ROW,\n",
+    "    output_file_name = output_file,\n",
+    "    retry_settings = BatchRetrySettings(max_retries=3, timeout=30),\n",
+    "    logging_level = logging.INFO,\n",
```
Comment (nick863) [inline sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb:749]: I am sorry, it looks like the backend do not like thid logging constant. Let us revert it to \"info\" and remove logging from imports.

Comment (SamGos93) [reply sdk/python/jobs/automl-standalone-jobs/automl-forecasting-task-bike-share/auto-ml-forecasting-bike-share.ipynb]: Reverted it back

**Review by novaturient95 — APPROVED** [review]

**Review by nick863 — APPROVED** [review]

**Pull request merged**